oshmem/mca/memheap/base/memheap_base_static.c

```diff
@@ -205,13 +217,31 @@ static int _check_pathname(struct map_segment_desc *seg)
     return OSHMEM_SUCCESS;
 }
 
+static memheap_static_segment_t *
+_allocate_static_segment(memheap_static_context_t *ctx)
+{
+    assert(ctx->n_segments <= ctx->capacity);
+
+    if (ctx->n_segments >= ctx->capacity) {
```

Comment:
can we reuse mca_memheap_base_allocate_segment?

Reply:
it conflicts with the assert above